src/main/kotlin/racing/controller/RacingControllerImpl.kt

```diff
@@ -0,0 +1,46 @@
+package racing.controller
+
+import racing.model.car.Car
+import racing.model.generator.NumberGenerator
+import racing.view.input.InputView
+import racing.view.result.ResultView
+
+internal class RacingControllerImpl(
+    private val inputView: InputView,
+    private val resultView: ResultView,
+    private val numberGenerator: NumberGenerator,
+) : RacingController {
+    override fun start() {
+        val carNames = inputView.promptAndValidateCarNamesInput()
+        val attemptCount = inputView.promptAndValidateAttemptCountInput()
+
+        val cars = getCarList(carNames)
+
+        resultView.printOutputTitle()
+        repeat(attemptCount) {
+            startRound(cars)
+            resultView.displayCarMovement(cars)
+        }
+        val raceWinners = getRaceWinners(cars)
+        resultView.displayRaceWinners(raceWinners)
+    }
+
+    private fun getCarList(carNames: List<String>): List<Car> {
+        return carNames.map { name -> Car(name = name) }
+    }
+
+    private fun startRound(
+        cars: List<Car>,
+        forwardLimit: Int = Car.DEFAULT_FORWARD_LIMIT,
+        numberGenerator: NumberGenerator = this.numberGenerator,
+    ) {
+        fun move(car: Car) = car.move(forwardLimit, numberGenerator)
+        cars.forEach(::move)
```

Comment:
함수를 인자로 받을 필요가 있을까요??

```suggestion
        cars.forEach { it.move(forwardLimit, numberGenerator)}
```

Reply:
괄호를 생략해서 구성하면 좀더 보기 쉽지 않을까 생각해서 시도 했었습니다.
이 부분은 제안주신 코드로 적용하겠습니다.